test/unit/custom_domain/CustomDomainLangsTest.php

```diff
@@ -18,14 +17,35 @@ protected function setUp()
             'foo.com/' => 'fr',
             'foo.com/path' => 'ja',
             'foo.com/dir/path' => 'zh-CHS',
-            'english.foo.com/' => 'en'
+            'english.foo.com/' => array('lang' => 'en', 'source' => 'global.foo.com/')
         );
         $this->customDomainLangs = new CustomDomainLangs($this->customDomainLangsSetting);
     }
 
-    private function getLang($customDomainlang)
+    private function getLang($customDomainLang)
     {
-        return $customDomainlang->getLang();
+        return $customDomainLang->getLang();
+    }
+
+    private function getHostAndPathWithoutTrailingSlash($customDomainLang)
+    {
+        return $customDomainLang->getHostAndPathWithoutTrailingSlash();
+    }
+
+    private function hashEquals($a, $b)
```

Comment:
Is there no php built-in function that can do this?

Reply:
Well there's the `array_diff_assoc` I can make use of. I guess I'll use that (but it still needs to be wrapped) since there's just no easy equality tests between associative arrays functioning as hash maps.

The original reason for why I hand coded this is I Googled "PHP hash map equals" and got the `hash_equals` built-in, and it's only available >5.6 so I decided against that. But ultimately it turns out it's not even doing what I expected, it literally compares the cryptographic hashes of two string?? D:

Reply:
in PHP parlance 'hash' is 'associative array' or 'multi-dimensional array' which is why you were getting results about string hashing. The hashes we deal with are only one level deep so we probably don't need a mega generic version.

Reply:
Yeah array_diff_assoc is clean enough, will use that.